Modules/timemodule.c

```diff
@@ -2390,3 +2399,53 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+#ifndef MS_WINDOWS
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+static int
+pysleep_zero_posix(void)
+{
+    assert(!PyErr_Occurred());
+    PyTime_t deadline, monotonic;
+    if (PyTime_Monotonic(&monotonic) < 0) {
+        return -1;
+    }
+    deadline = monotonic;
+    do {
+        int ret, err;
+        // POSIX-compliant select(2) allows the 'timeout' parameter to
+        // be modified but also mandates that the function should return
+        // immediately if *both* structure's fields are zero (which is
+        // the case here).
+        //
+        // However, since System V (but not BSD) variant typically sets
+        // the timeout before returning (but does not specify whether
+        // this is also the case for zero timeouts), we prefer supplying
+        // a fresh timeout everytime.
+        struct timeval zero = {0, 0};
+        Py_BEGIN_ALLOW_THREADS
+        ret = select(0, (fd_set *)0, (fd_set *)0, (fd_set *)0, &zero);
+        err = errno;
```

Comment:
nit: I think we can get rid of `err`, seems like an artifact of the version of this code that has a bunch of #ifdef